landlab/ca/cfuncs.pyx

```diff
@@ -428,11 +420,12 @@ cpdef get_next_event(
 @cython.boundscheck(True)
 @cython.wraparound(False)
 cpdef get_next_event_new(
-    DTYPE_INT_t link, DTYPE_INT_t current_state,
-    DTYPE_t current_time,
-    np.ndarray[DTYPE_INT_t, ndim=1] n_trn,
-    np.ndarray[DTYPE_INT_t, ndim=2] trn_id,
-    np.ndarray[DTYPE_t, ndim=1] trn_rate,
+    long link,
+    long current_state,
+    double current_time,
+    id_t [:] n_trn,
+    id_t [:, :] trn_id,
+    cython.floating [:] trn_rate,
```

Comment:
We may not need a fused type here since this function may never be called with an array for 32bit floats. The fused type here just makes it work with 16, 32, 64 (and 128?) bit floats.